### DISCUSSION TITLE
async

### DISCUSSION THREAD

**Discussion by pouemes** [Leaflet/Leaflet#8241] [Q&A] [answered]
Hello all
i would to load leaflet.js and a second javascript with contain my map like this, both in async
is it possible and how?
a great thank
`var map=L.map("map",{zoomControl:false,fullscreenControl:{pseudoFullscreen:true},minZoom:2,maxZoom:12,maxBounds:[[-90,-1/0],[90,1/0]],worldCopyJump:!0,scrollWheelZoom:"center",doubleClickZoom:"center",touchZoom:"center"});
var Url="https://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{y}/{x}";
var tile=L.tileLayer(Url,{attribution:"Powered by Esri | Esri, HERE, Garmin, USGS, NGA"}).addTo(map);
map.setView(llng,zo);
var json={"type":"FeatureCollection","features":[
{"type":"Feature","features":{"racine":"xxxx","location":"yyy","name":"zzz"},"geometry":{"type":"Point","coordinates":[44.609,32.611]}},

etc etc`

**Comment by Malvoz** [comment, marked as answer]
This has already been answered on SO: https://stackoverflow.com/a/45282437/9314312.

It'd be good to document this in future [performance best practices](https://github.com/Leaflet/Leaflet/issues/4239#:~:text=Performance%20best%20practices).

**Comment by pouemes** [comment]
thanks malvoz

i have also tried like this but dont know if it's correct

<html>
  <head>
    <title>Eine OSM Karte mit Leaflet</title>
    <link rel="stylesheet" href="https://unpkg.com/leaflet/dist/leaflet.css" />
    <script src="https://unpkg.com/leaflet/dist/leaflet.js" async></script>
  </head>
<body>
  <div id="map" style="width: 600px; height: 400px"></div>
  <script>
  new Promise(function (resolve, reject) {
(function promise(){
if (typeof L !== "undefined"  && typeof map !== "undefined") {
return resolve();
}
setTimeout(promise, 300); }());
}).then(function(){
map();
});
function map() {
    var map = L.map('map',
      {
        center: [50.27264, 7.26469],
        zoom: 10
      }
    );
    L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png').addTo(map);
  };
  </script>
</body>
</html>
